Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1123 @@
+                {
+                    $nonCompliantFrontDoorEPs += $frontDoorResource
+
+                }
+                
+
+              }
+
+            }
+            catch
+            {
+                $frontDoorsSkipped += $frontDoorResource
+                Write-Host "Error fetching Front Door configuration: Resource Name: [$($resourceName)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
+                $logSkippedResources += $logResource
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    
+
+   foreach($item in $nonCompliantFrontDoorEPs)
+   {
+        if ($item.MinimumTlsVersion -eq $requiredMinTLSVersion)
+        {
+           Write-Host "Minimum TLS Version is set on the custom domains of front doors." -ForegroundColor $([Constants]::MessageType.Update)
+           Write-Host "Skipping this Front Door..." -ForegroundColor $([Constants]::MessageType.Warning)
+           Write-Host $([Constants]::SingleDashLine)
+           $logResource = @{}
+           $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+           $logResource.Add("ResourceName",($_.ResourceName))
+           $logResource.Add("Reason","Minimum TLS Version is set on all the custom domains of front door.")    
+           $logSkippedResources += $logResource
+        }
+        else 
+        {
+            if (-not $item.MinimumTlsVersion -ne $requiredMinTLSVersion)
+            {
+                Write-Host "Minimum TLS Version is not set on the custom domain" $item.DomainName  -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            $NonCompliantFrontDoorEndpoints += $item | Select-Object @{N='ResourceGroupName';E={$item.ResourceGroupName}},
+                                                                                            @{N='ResourceName';E={$item.ResourceName}},
+                                                                                            @{N='DomainName';E={$item.DomainName}},
+                                                                                            @{N='CertificateType';E={$item.CertificateType}},
+                                                                                            @{N='MinimumTlsVersion';E={$item.MinimumTlsVersion}},
+                                                                                            @{N='isMinTLSVersionSetOnCustomDomain';E={$item.isMinTLSVersionSetOnCustomDomain}}
+        }
+    }
```

Comment:
This code block from 495 to 522 can be avoided.

Reply:
Removed.